plugins/processors/topk/topk.go

```diff
@@ -0,0 +1,402 @@
+  # This effectively means that more than K metrics may be returned. If you need to return only the top k metrics regardless of grouping, use the simple_topk setting
+  fields = ["memory_rss"]      # Over which fields are the top k are calculated. Default: ["value"]
+  aggregation = "avg"          # What aggregation to use. Default: "avg". Options: sum, avg, min, max
+
+  bottomk = false              # Instead of the top k largest metrics, return the bottom k lowest metrics. Default: false
+  simple_topk = false          # If true, this will override any GroupBy options and assign each metric its own individual group. Default: false
+  drop_no_group = true         # Drop any metrics that do fit in any group (due to nonexistent tags). Default: true
+  drop_non_top = true          # Drop the metrics that do not make the cut for the top k. Default: true
+
+  group_by_tag = ""            # The plugin assigns each metric a GroupBy tag generated from its name and tags.
+                               # If this setting is different than "" the plugin will add a tag (which name will be the value of
+                               # this setting) to each metric with the value of the calculated GroupBy tag.
+                               # Useful for debugging. Default: ""
+
+  position_field = ""          # This settings provides a way to know the position of each metric in the top k.
+                               # If set to a value different than "", then a field (which name will be prefixed with the value of this setting) will be
+                               # added to each every metric for each field over which an aggregation was made.
+                               # This field will contain the ranking of the group that the metric belonged to. When aggregating
+                               # over several fields, several fields will be added (one for each field over which an aggregation was calculated).
+                               # Useful for debugging. Default: ""
+
+  aggregation_field = ""       # This setting provies a way know the what values the plugin is generating when aggregating the fields
+                               # If set to a value different than "", then a field (which name will be prefixed with the value of this setting) will be added
+                               # to each metric which was part of a field aggregation.
+                               # The value of the added field will be the value of the result of the aggregation operation for that metric's group. When aggregating
+                               # over several fields, several fields will be added (one for each field over which an aggregation was calculated).
+                               # Useful for debugging.Default: ""
+`
+
+type MetricAggregation struct {
+	groupbykey string
+	values     map[string]float64
+}
+
+type Aggregations struct {
+	metrics []MetricAggregation
+	field   string
+}
+
+func (ags Aggregations) Len() int {
```

Comment:
I can't. The sort interface asks for a direct value:
```
plugins/processors/topk/topk.go:165: cannot use aggs (type Aggregations) as type sort.Interface in argument to sort.Sort:
	Aggregations does not implement sort.Interface (Len method has pointer receiver)
plugins/processors/topk/topk.go:167: cannot use aggs (type Aggregations) as type sort.Interface in argument to sort.Reverse:
	Aggregations does not implement sort.Interface (Len method has pointer receiver)
```

Reply:
Okay, BTW I have been heard from my colleagues that it is now preferred to use `sort.SliceStable` with a less func instead of defining ordering on the type.